src/cdc/runner.zig

```diff
@@ -869,22 +869,24 @@ const Metrics = struct {
     const TimingSummary = struct {
         timer: std.time.Timer,
 
-        duration_min_ms: ?u64 = null,
-        duration_max_ms: ?u64 = null,
-        duration_sum_ms: u64 = 0,
+        duration_min_ns: ?u64 = null,
+        duration_max_ns: ?u64 = null,
+        duration_sum_ns: u64 = 0,
         event_count: u64 = 0,
         count: u64 = 0,
 
         fn record(
             metrics: *TimingSummary,
             event_count: u64,
         ) void {
-            const duration_ms: u64 = @divFloor(metrics.timer.read(), std.time.ns_per_ms);
-
-            metrics.duration_min_ms =
-                @min(duration_ms, metrics.duration_min_ms orelse std.math.maxInt(u64));
-            metrics.duration_max_ms = @max(duration_ms, metrics.duration_max_ms orelse 0);
-            metrics.duration_sum_ms += duration_ms;
+            const duration_ns: u64 = metrics.timer.read();
+            assert(duration_ns > 0);
```

Comment:
I think this should be `maybe`: timer.read() doesn't have a contract that it is _strictly_ monotonic. And I think this might even fire in practice? While the return value is ns precision, the underlying clock _could_ be less precise. Imagine we are running in a web-assembly which applies side-channel attack mitigations and makes clocks really coarse-grained? 

So let's do defensive thing here and _not_ assume that time flows with positive speed!

Reply:
Tbh, it feels fishy that we get a collision here with ms precision, could it be that we have a logical bug somewhere else? Though, I did a rather thorough reading of the code around, and it feels fine!

Reply:
> But we did call reset, no?

Yeah we do, sorry, apparently I couldn't read my own code :facepalm: 

> In general, not a fan of std.time.Timer API! Consider doing something like

Don't want to deviate too much from the similar implementation already in place in the StateMachine. We should replace both soon.